test/mpi.py

```diff
@@ -23,36 +23,251 @@
 # along with vsc-mympirun.  If not, see <http://www.gnu.org/licenses/>.
 #
 """
-Tests for the vsc.utils.missing module.
+Tests for the vsc.mympirun.mpi.mpi module.
 
-@author: Jens Timmerman (Ghent University)
+@author: Jeroen De Clerck
 """
 import os
 import stat
-from unittest import TestCase
+import unittest
+import re
+import string
 
+from vsc.utils.run import run_simple
 from vsc.mympirun.factory import getinstance
-from vsc.mympirun.mpi.mpi import MPI
+import vsc.mympirun.mpi.mpi as mpim
+#from vsc.mympirun.mpi.openmpi import OpenMPI
 from vsc.mympirun.rm.local import Local
 from vsc.mympirun.option import MympirunOption
 
 # we wish to use the mpirun we ship
 os.environ["PATH"] = os.path.dirname(os.path.realpath(__file__)) + os.pathsep + os.environ["PATH"]
 
 
-class TestMPI(TestCase):
-    """Test for the mpi class."""
+class TestMPI(unittest.TestCase):
 
-    def test_MPI_local(self):
-        """"Test the MPI class with the local scheduler"""
-        # options
-        # m = MympirunOption()
-        # m.args = ['echo', 'foo']
-        # # should not throw an error
-        # mpi_instance = getinstance(MPI, Local, m)
-        # mpi_instance.main()
+    """tests for vsc.mympirun.mpi.mpi functions"""
 
-        # # check for correct .mpd.conf file
-        # mpdconffn = os.path.expanduser('~/.mpd.conf')
-        # perms = stat.S_IMODE(os.stat(mpdconffn).st_mode)
-        # self.assertEqual(perms, 0400, msg='permissions %0o for mpd.conf %s' % (perms, mpdconffn))
+    #######################
+    ## General functions ##
+    #######################
+
+    def test_what_mpi(self):
+        """test if what_mpi returns the correct mpi flavor"""
+        scriptnames = ["ompirun", "mpirun", "impirun", "mympirun"]
+        for scriptname in scriptnames:
+            # if the scriptname is an executable located on this machine
+            if mpim.which(scriptname):
```

Comment:
it's easy to add more mpirun aliasses the the scriptname list. If the which() is dropped you'd have to dig in the code and make sure that the scriptname list only contains the ones that are installed on the system.